iModelCore/ecobjects/src/SchemaMerger.cpp

```diff
@@ -352,6 +352,19 @@ BentleyStatus SchemaMerger::MergeSchemas(SchemaMergeResult& result, bvector<ECSc
         {
         DumpSchemasToFile(result.GetResults(), dumpLocation.c_str(), "Result");
         }
+    
+    if(false == options.GetSkipValidation())
+        {
+        for(const auto& schema : result.GetResults())
+            {
+            if(!const_cast<ECSchemaP>(schema)->Validate(true))

```

Comment:
you can use m_modifiedSchemas from results here then the const cast is not needed.